contracts/loan_manager/src/positions.rs

```diff
@@ -1,50 +1,25 @@
 use crate::storage_types::{
     Loan, LoansDataKey, POSITIONS_BUMP_AMOUNT, POSITIONS_LIFETIME_THRESHOLD,
 };
-use soroban_sdk::{Address, Env, IntoVal, Val};
-
-pub fn init_loan(
-    e: &Env,
-    addr: Address,
-    borrowed_amount: i128,
-    borrowed_from: Address,
-    collateral_amount: i128,
-    collateral_from: Address,
-    health_factor: i128,
-) {
-    write_positions(
-        e,
-        addr,
-        borrowed_amount,
-        borrowed_from,
-        collateral_amount,
-        collateral_from,
-        health_factor,
-    );
+use soroban_sdk::{Address, Env};
+
+pub fn init_loan(e: &Env, addr: Address, loan: Loan) {
+    write_positions(e, addr, loan);
 }
 
-fn write_positions(
-    e: &Env,
-    addr: Address,
-    borrowed_amount: i128,
-    borrowed_from: Address,
-    collateral_amount: i128,
-    collateral_from: Address,
-    health_factor: i128,
-) {
+fn write_positions(e: &Env, addr: Address, loan: Loan) {
     let key = LoansDataKey::Loan(addr);
 
-    let loan: Loan = Loan {
-        borrowed_amount,
-        borrowed_from,
-        collateral_amount,
-        collateral_from,
-        health_factor,
+    let writeable_loan: Loan = Loan {
+        borrowed_amount: loan.borrowed_amount,
+        borrowed_from: loan.borrowed_from,
+        collateral_amount: loan.collateral_amount,
+        collateral_from: loan.collateral_from,
+        health_factor: loan.health_factor,
+        unpaid_interest: loan.unpaid_interest,
```

Comment:
Miksi näin pitää tehdä? Eikö voisi suoraan antaa `&loan` parametriksi `storage().set(...)`:lle?

Reply:
Joo voi ja korjaan. Mul oli joku aivopieru täs et siin ei tehtäis mitään jos en tee noin.. :man_facepalming: